File: CS/FileTree/Imported/Win32FunctionService.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Runtime.InteropServices;

namespace OpenDialogTest.Imported
{
    public class Win32FunctionService
    {
        public static Image GetFileSystemImage(string fileName)
        {
            SHFILEINFO shinfo = new SHFILEINFO();
            Win32.SHGetFileInfo(fileName, 0, ref shinfo, (uint)Marshal.SizeOf(shinfo), Win32.SHGFI_ICON | Win32.SHGFI_SMALLICON);
            Image image = Icon.FromHandle(shinfo.hIcon).ToBitmap();
            return image;
        }
    }
}

File: VB/FileTree/Imported/Win32FunctionService.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Runtime.InteropServices

Namespace OpenDialogTest.Imported
	Public Class Win32FunctionService
		Public Shared Function GetFileSystemImage(ByVal fileName As String) As Image
			Dim shinfo As New SHFILEINFO()
			Win32.SHGetFileInfo(fileName, 0, shinfo, CUInt(Marshal.SizeOf(shinfo)), Win32.SHGFI_ICON Or Win32.SHGFI_SMALLICON)
			Dim image As Image = Icon.FromHandle(shinfo.hIcon).ToBitmap()
			Return image
		End Function
	End Class
End Namespace
